src/transformers/pipelines/fill_in_middle.py

```diff
@@ -0,0 +1,333 @@
+import enum
+import warnings
+from collections import OrderedDict
+
+from ..utils import add_end_docstrings, is_tf_available, is_torch_available
+from .base import PIPELINE_INIT_ARGS, Pipeline
+
+
+class ReturnType(enum.Enum):
+    TENSORS = 0
+    NEW_TEXT = 1
+    FULL_TEXT = 2
+
+
+if is_torch_available():
+    FIM_SUPPORTED_MODELS_TORCH = [
```

Comment:
yep this sounds alright to me. The mapping could just check if the model name supports it, as some model do / some don't but it could be a bit too restrictive. We can just check if the model type is in the AUTO_MODEL_FOR_CONDITIONAL_GENERATION + if it has the necessary tokens added to the vocab (base the check on models that currently support this + incentives people to use the same code notation for the tokens that serve as prefix / suffix etc

Reply:
Is `AUTO_MODEL_FOR_CONDITIONAL_GENERATION` in `modeling_auto.py`? I am unable to find it.